src/java.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * sonar-scanner-npm
+ * Copyright (C) 2022-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { log, LogLevel } from './logging';
+import axios from 'axios';
+import semver, { SemVer } from 'semver';
+import { SONARQUBE_JRE_PROVISIONING_MIN_VERSION } from './constants';
+import { downloadFile } from './download';
+import { JreMetaData, PlatformInfo } from './types';
+
+type Parameters = Record<string, string | undefined>;
+
+function isSonarQube(parameters: Parameters): boolean {
+  if (parameters['sonar.scanner.sonarcloudUrl'] !== undefined) {
+    return false;
+  }
+  if (parameters['sonar.host.url'] !== undefined) {
```

Comment:
What if `parameters['sonar.host.url'] === 'https://sonarcloud.io`? In this case, we should return `false`